grafana-plugin/src/models/alert_receive_channel/alert_receive_channel.helpers.ts

```diff
@@ -1,11 +1,172 @@
-import { AlertReceiveChannel } from './alert_receive_channel.types';
+import { ChannelFilter } from 'models/channel_filter/channel_filter.types';
+import { GrafanaTeam } from 'models/grafana_team/grafana_team.types';
+import { makeRequest } from 'network/network';
+import { ApiSchemas } from 'network/oncall-api/api.types';
+import { onCallApi } from 'network/oncall-api/http-client';
+import { SelectOption } from 'state/types';
+import { showApiError } from 'utils/utils';
 
-export function getAlertReceiveChannelDisplayName(alertReceiveChannel?: AlertReceiveChannel, withDescription = true) {
-  if (!alertReceiveChannel) {
-    return '';
+import { AlertReceiveChannelStore } from './alert_receive_channel';
+import { MaintenanceMode } from './alert_receive_channel.types';
+
+export class AlertReceiveChannelHelper {
+  static getAlertReceiveChannelDisplayName(
+    alertReceiveChannel?: ApiSchemas['AlertReceiveChannel'],
+    withDescription = true
+  ) {
+    if (!alertReceiveChannel) {
+      return '';
+    }
+
+    return withDescription && alertReceiveChannel.description
+      ? `${alertReceiveChannel.verbal_name} (${alertReceiveChannel.description})`
+      : alertReceiveChannel.verbal_name;
   }
 
-  return withDescription && alertReceiveChannel.description
-    ? `${alertReceiveChannel.verbal_name} (${alertReceiveChannel.description})`
-    : alertReceiveChannel.verbal_name;
+  static getSearchResult = (store: AlertReceiveChannelStore) =>
+    store.searchResult
+      ? store.searchResult.map(
+          (alertReceiveChannelId: ApiSchemas['AlertReceiveChannel']['id']) => store.items?.[alertReceiveChannelId]
+        )
+      : undefined;
+
+  static getPaginatedSearchResult = (store: AlertReceiveChannelStore) =>
+    store.paginatedSearchResult
+      ? {
+          page_size: store.paginatedSearchResult.page_size,
+          count: store.paginatedSearchResult.count,
+          results: store.paginatedSearchResult.results?.map(
+            (alertReceiveChannelId: ApiSchemas['AlertReceiveChannel']['id']) => store.items?.[alertReceiveChannelId]
+          ),
+        }
+      : undefined;
+
+  static getIntegration = (
+    store: AlertReceiveChannelStore,
+    alertReceiveChannel: Partial<ApiSchemas['AlertReceiveChannel']>
+  ): SelectOption => {
+    return (
+      store.alertReceiveChannelOptions &&
+      alertReceiveChannel &&
+      store.alertReceiveChannelOptions.find(
+        (alertReceiveChannelOption: SelectOption) => alertReceiveChannelOption.value === alertReceiveChannel.integration
+      )
+    );
+  };
+
+  static deleteAlertReceiveChannel = async (id: ApiSchemas['AlertReceiveChannel']['id']) =>
+    (await onCallApi().DELETE('/alert_receive_channels/{id}/', { params: { path: { id } } })).data;
+
+  static getGrafanaAlertingContactPoints = async () =>
+    (await onCallApi().GET('/alert_receive_channels/contact_points/', undefined)).data;
```

Comment:
can we make the second param optional to avoid passing `undefined`?

Reply:
Unfortunately not yet, I think it should be fixed in the next version of the lib. For now we need to pass it